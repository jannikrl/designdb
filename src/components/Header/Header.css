.header {
    height: 90px;
    width: 100%;
}

.header .logo {
    float: left;
}

.header .logo a {
    text-decoration: none;
    color: black;
}

.header nav {
    float: right;
}