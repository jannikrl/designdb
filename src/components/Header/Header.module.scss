.header {
  height: 90px;
  width: 100%;

  .logo {
    float: left;

    a {
      text-decoration: none;
      color: black;
    }
  }

  nav {
    float: right;
  }
}