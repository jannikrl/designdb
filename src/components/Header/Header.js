import React from "react";
import { Link } from "react-router-dom";
import "./Header.css";

const Header = (props) => {
  const loginLink = !props.isAuthenticated ? (
    <Link to="/auth">Login</Link>
  ) : null;
  const logoutLink = props.isAuthenticated ? (
    <Link to="/logout">Logout</Link>
  ) : null;

  return (
    <header className={"header"}>
      <h1 className={"logo"}><Link to="/">designdb</Link></h1>
      <nav>
        {loginLink}
        {logoutLink}
      </nav>
    </header>
  );
};

export default Header;
