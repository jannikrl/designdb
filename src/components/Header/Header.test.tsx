import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Header from "./Header";

it("renders correctly", () => {
  const wrapper = shallow(<Header isAuthenticated />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
 