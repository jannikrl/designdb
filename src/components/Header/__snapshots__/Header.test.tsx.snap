// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="header"
>
  <h1
    className="logo"
  >
    <Link
      to="/"
    >
      Desssign
    </Link>
  </h1>
  <nav>
    <div>
      <Link
        to="/design/create"
      >
        New design
      </Link>
      <Link
        to="/designer/create"
      >
        New designer
      </Link>
      <Link
        to="/manufacturer/create"
      >
        New manufacturer
      </Link>
      <Link
        to="/logout"
      >
        Logout
      </Link>
    </div>
  </nav>
</header>
`;
